### PR TITLE
[CARBONDATA-1166] Fix bug on decimal partition column and add test case for all primitive column

### PR DESCRIPTION
**Pull request by QiangCai**
1. fix bug on decimal partition column

2. add test case for all primitive column
support datatype: smallInt, int, bigInt, float, double, decimal, timestamp, date, string, varchar, char

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2428/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/307/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/307/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/307/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.dataretention/DataRetentionConcurrencyTestCase/DataRetention_Concurrency_load_date/'><strong>org.apache.carbondata.spark.testsuite.dataretention.DataRetentionConcurrencyTestCase.DataRetention_Concurrency_load_date</strong></a></li></ul>

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2430/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/309/

**Comment by QiangCai** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/360/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2475/

**Comment by QiangCai** [comment]
@gvramana

File: core/src/main/java/org/apache/carbondata/core/scan/filter/partition/InFilterImpl.java [hunk 1]
```diff
@@ -48,6 +50,9 @@ public InFilterImpl(InExpression in, PartitionInfo partitionInfo) {
       Object value = PartitionUtil.getDataBasedOnDataTypeForFilter(
           literal.getLiteralExpValue().toString(),
           partitionInfo.getColumnSchemaList().get(0).getDataType());
+      if (PartitionType.RANGE == partitionInfo.getPartitionType() && value instanceof String) {
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/partition/InFilterImpl.java]: for range partition, we need compare string value.
hash and list no need

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**